src/GovUk.Education.ExploreEducationStatistics.Admin/Services/UpdateChartFilesService.cs

```diff
@@ -0,0 +1,168 @@
+using GovUk.Education.ExploreEducationStatistics.Common.Model.Chart;
+using GovUk.Education.ExploreEducationStatistics.Content.Model;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.Storage;
+using Microsoft.Azure.Storage.Blob;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using static GovUk.Education.ExploreEducationStatistics.Common.BlobContainerNames;
+using static GovUk.Education.ExploreEducationStatistics.Common.Services.FileStoragePathUtils;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Services
+{
+    public class UpdateChartFilesService : IUpdateChartFilesService
+    {
+        private readonly ContentDbContext _contentDbContext;
+        private readonly ILogger _logger;
+
+        private readonly string _storageConnectionString;
+        
+        public UpdateChartFilesService(
+            IConfiguration configuration,
+            ContentDbContext contentDbContext,
+            ILogger<UpdateChartFilesService> logger)
+        {
+            _storageConnectionString = configuration.GetValue<string>("CoreStorage");
+            _contentDbContext = contentDbContext;
+            _logger = logger;
+        }
+
+        public async Task<Either<ActionResult, bool>> UpdateChartFiles()
+        {
+            var refs = _contentDbContext
+                .ReleaseFileReferences
+                .Where(rfr => rfr.ReleaseFileType == ReleaseFileTypes.Chart)
+                .ToList();
+
+            var result = await UpdateChartFilesAsync(refs);
+
+            if (result.IsLeft)
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        private async Task<Either<ActionResult, bool>> UpdateChartFilesAsync(List<ReleaseFileReference> chartReferences)
+        {
+            var blobContainer = await GetCloudBlobContainer();
+
+            foreach (var fileReference in chartReferences)
+            {
+                var oldName = fileReference.Filename;
+                var oldBlobPath = AdminReleasePathWithFilename(fileReference.ReleaseId, oldName);
+
+                // Files should exists first time this is run in storage
+                if (blobContainer.GetBlockBlobReference(oldBlobPath).Exists())
+                {
+                    if (await CopyBlob(fileReference, blobContainer, oldBlobPath, fileReference.Id))
+                    {
+                        // Change ref to any datablocks with charts where used
+
+                        await UpdateDataBlock(fileReference, fileReference.Id);
+
+                        await _contentDbContext.SaveChangesAsync();
+
+                        _logger.LogInformation(
+                            $"Update blob: {fileReference.ReleaseId}:{fileReference.Filename} successfully");
+                    }
+                }
+            }
+
+            return true;
```

Comment:
We now have a `Unit.Instance` type can return instead (represents a generic `void` type).

Reply:
done